aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -170,6 +171,17 @@
    */
   public static final String S3FILEIO_SESSION_TOKEN = "s3.session-token";
 
+  /**
+   * Enable to make S3FileIO, to make cross-region call to the region specified in the ARN.
+   * <p>
+   * When set, the client can make a cross-region call to the region specified in the ARN,
+   * otherwise an exception will be thrown if cross-region call is tried.
+   *
+   * For more details see : https://github.com/aws/aws-sdk-java-v2/blob/master/services/s3/src/main/java/software/amazon/awssdk/services/s3/S3Configuration.java#L186
```

Comment:
line number might change, linking to the github code is probably a bad idea, sorry for the bad suggestion. We can link to this: https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/s3/S3Configuration.html#useArnRegionEnabled--